lib/linklist.c

```diff
@@ -242,6 +288,23 @@ void list_delete_all_node(struct list *list)
 	list->count = 0;
 }
 
+void list_cond_delete_nodes(struct list *list, cond_del_fn_t  cond)
+{
+	struct listnode *node;
+	struct listnode *next;
+
+	assert(list);
+
+	for (node = list->head; node; node = next) {
```

Comment:
We have a macro for this: `for (ALL_LIST_ELEMENTS(list, node, next, data))`
Please use it

Reply:
Will do that